### PR TITLE
[Raiden Weight Sync 2/7] Support inhomogeneous layer cycle interval in raiden_unscan

### PR DESCRIPTION
**Pull request by YixuanWang-99**
## Overview
Part of the stacked Raiden weight-sync enablement PR chain replacing #5089.

**Stack:**
- [M7] https://github.com/AI-Hypercomputer/maxtext/pull/5166: Cross-repo drift guard
- **[M1] AI-Hypercomputer/maxtext (this PR)**: Inhomogeneous layer cycle support in raiden_unscan
- [M2] AI-Hypercomputer/maxtext: MoE 128-lane layout and padding (pairs with Tunix T5)
- [M3] AI-Hypercomputer/maxtext: Target-free streaming weight conversion
- [M5] AI-Hypercomputer/maxtext: Shared expert gate weight_dtype fix
- [M4] AI-Hypercomputer/maxtext: TrainingEngine Raiden FFI integration (pairs with Tunix T2a)
- [M6] AI-Hypercomputer/maxtext: Rollout cleanup

## Details
- Supports `cycle_interval` in `unscan_layers` for models with periodic inhomogeneous layers (e.g. Qwen3.5 GDN).
- **Review thread addressed**: When `slot is None` and `cycle_interval > 1`, raises a descriptive `ValueError` naming the key and the missing cycle-slot prefix instead of misleadingly attributing the failure to `num_layers`.
- Adds unit tests in `tests/unit/raiden_unscan_test.py`.

## Verification
- `pytest tests/unit/raiden_unscan_test.py` (11/11 passed).

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run end-to-end tests tests and provided workload links above if applicable.
- [x] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

**Review by igorts-git — APPROVED** [review]
please address review bot comments. I think they are meaningful.

File: tests/unit/raiden_unscan_test.py [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+class UnscannedModel(nnx.Module):
+  """A sampler-side model: one submodule per layer, named `layers_0..N-1`."""
+
+  def __init__(self, num_layers: int = _NUM_LAYERS):
+    for i in range(num_layers):
+      setattr(self, f"layers_{i}", UnscannedInner())
+    self.embed = nnx.Param(jnp.zeros((_VOCAB, _IN)))
+
+
+class UnscanLayersTest(absltest.TestCase):
```
Comment (khatwanimohit) [inline tests/unit/raiden_unscan_test.py:90]: can you add a test which tests unscan_layers with cycle_interval > 1

Comment (khatwanimohit) [reply tests/unit/raiden_unscan_test.py:90]: done

**Review by khatwanimohit — APPROVED** [review]
